sbaid/model/simulator/dummy/dummy_simulator.py

```diff
@@ -24,34 +97,92 @@ def route(self) -> Gio.ListModel:
+                                                                 CrossSectionType.COMBINED,
+                                                                 Location(0, 0), max_lanes,
+                                                                 False))
+                self._sequence[int(snapshot_time)] = current_input
+
+        self._simulation_start_time = min(self._sequence.keys())
+        self._simulation_end_time = max(self._sequence.keys())
 
     async def create_cross_section(self, location: Location,
                                    cross_section_type: CrossSectionType) -> int:
-        """TODO"""
-        return 0
+        """Has no effect. Raises an exception."""
+        raise NotSupportedException("The dummy simulator does not support creating"
+                                    "new cross sections.")
 
     async def remove_cross_section(self, cross_section_id: str) -> None:
-        """TODO"""
+        """Has no effect. Raises an exception."""
+        raise NotSupportedException("The dummy simulator does not support removing cross sections.")
 
     async def move_cross_section(self, cross_section_id: str, new_location: Location) -> None:
-        """TODO"""
+        """Has no effect. Raises an exception."""
+        raise NotSupportedException("The dummy simulator does not support moving cross sections.")
 
     async def init_simulation(self) -> tuple[GLib.DateTime, int]:
-        return GLib.DateTime.new_now_local(), 0
+        """Initialize the simulator. Always returns 0 as the starting point
+        and the amount of input measurements as the simulation length."""
+        self._pointer = 0
+        return GLib.DateTime.new_now_local(), self._simulation_end_time
 
     async def continue_simulation(self, span: int) -> None:
-        """TODO"""
+        """Set the measuring pointer to the given location in the measurements."""
+        if (self._pointer + span) <= self._simulation_end_time:
+            self._pointer = self._pointer + span
+        else:
+            raise EndOfSimulationException(f"The simulation cannot be continued {span} steps.")
 
     async def measure(self) -> Input:
-        return None
+        """Return the current input of the simulator.
+        If there is no measurement given at the current time,
+        the nearest neighbouring measurement is returned."""
+        if not self._sequence:
+            raise FileNotFoundError("No simulator file has been loaded.")
+        if not self._sequence.keys().__contains__(self._pointer):
+            nearest = min(self._sequence.keys(), key=lambda x: abs(x - self._pointer))
+            return self._sequence[nearest]
+        return self._sequence[self._pointer]
 
     async def set_display(self, display: Display) -> None:
-        """TODO"""
+        """Has no effect. Raises an exception."""
+        raise NotSupportedException("The dummy simulator does not support removing cross sections.")
```

Comment:
hier würd ichs nicht machen weil wir das hier ja während der simulation laufen haben und dementsprechend nicht den user informieren und die effects dieser methode ja implementationsspezifisch sind (lediglich die measurements ändern sich (theoretisch) aber in einer unpredicitable way für den nutzer)